protocol/0091-LIQM-liquidity_mechanisms.md

```diff
@@ -0,0 +1,118 @@
+
+> [!NOTE]
+> the maker fees is scaled by the assets quantum such that 1 ELS point is earned for approx. every 1 USD of maker fees received.
+
+## Designating Liquidity Providers
+
+A party will only be able to receive liquidity fees or earn liquidity rewards providing they are designated as an LP for that epoch.
+
+A party will only be designated as an LP providing they received more than a specified proportion of the markets total maker fees in the previous epoch, let this requirement be $N$ (the network parameter `liquidityProvider.proportionRequirement`). Throughout the epoch the network will track each parties maker fees, $M$.  At the end of epoch $i$, a party will be designated as an LP for epoch $i+1$ providing:
+
+$$\frac{V_{i_j}}{\sum_{k}^{n}{M_{i_j}}} >= N$$
+
+Where:
+
+- $M_{i_j}$ is the maker fees of party ${j}$ in epoch ${i}$
+- $N$ the requirement specified by the network parameter `liquidityProvider.proportionRequirement` 
+
+By designating parties as LPs on an epoch by epoch basis the protocol ensures:
+
+- "in-active" parties with a large number of ELS points will no longer be rewarded through liquidity mechanisms should they stop providing liquidity (or provide uncompetitive liquidity).
+- "late-arriving" parties with a small number of ELS points will be rewarded through liquidity mechanisms if they provide competitive liquidity and as such comprise a larger proportion of the markets volume.
+
+
+## Implied Commitment Amount
+
+Each LPs `implied commitment amount` is defined as, the maximum volume of notional liquidity that they supplied within a specified range for at least N % of the epoch (where the liquidity range and N are market configurable parameters).
+
+To calculate the implied commitment amount, throughout the epoch, the network must sample and store the volume of notional liquidity supplied by each LP at that point. For now this is done once a block but could be sampled randomly to reduce the amount of data stored.
+
+### Instantaneous Supplied Liquidity
+
+Calculating the liquidity supplied at any given point in time is done as follows:
+
+Whilst in continuous trading:
+
+- If there is no mid price each LP is treated as supplying `0` liquidity.
+
+- If there is a mid price calculate the volume of notional that is in the range.
+
+```text
+(1.0-market.liquidity.priceRange) x mid <= price levels <= (1+market.liquidity.priceRange)x mid.
+```
+
+Whilst in monitoring auctions:
+
+- If there is an indicative uncrossing price calculate the volume of notional that is in the range.
+
+```text
+(1.0-market.liquidity.priceRange) x min(last trade price, indicative uncrossing price) <=  price levels <= (1.0+market.liquidity.priceRange) x max(last trade price, indicative uncrossing price).
+```
+
+- If there is no 'indicative uncrossing price' then volume placed at any price should count towards the LP's commitment i.e the price range is interpreted as
+
+```text
+-infinity <=  price levels <= infinity
```

Comment:
Perhaps we could narrow it down to the auction uncrossing range (used to calculate the uncrossing price) or some stretched version of it and calculate it retrospectively only once the auction uncrosses? Though I guess that's potentially more data to be stored in the interim.

Reply:
Auction uncrossing range is a good idea, in hind sight not a massive fan of infinity as an LP can game it by placing small orders at large prices during auctions when there is no uncrossing price.

Do we have an uncrossing range when the book isn't crossed though?

Reply:
We don't, though I was thinking just `0` commitment in that case. So LPs only get commitment implied if they successfully contributed to auction having uncrossed with non-zero volume.